### PR TITLE
feat(function): new stuff

### PR DESCRIPTION
**Pull request by Mte90**
Adding the failable function and default values for function parameters

### CODE REVIEW

File: docs/basic_syntax/functions.md [hunk 1]
```diff
@@ -27,6 +27,16 @@ fun myFunction(arg1: Num, arg2: Num): Num {
...
 An interesting fact about functions is that they are not parsed unless they are used. This behavior exists because Amber allows you to omit specifying any type at all. When you use such function - then it generates different variants of this function with types that were used (without any duplications).
 
+You can use default parameters but requires to define the type of the parameter:
+
+```ab
+fun addition(a: Num, b: Num = 100): Num {
+    return a + b
+}
+echo addition(10)
+// Outputs: 110
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:37]:
```suggestion
echo addition(10) // Outputs: 110
echo addition(10, 20) // Outputs: 30
```

File: docs/basic_syntax/functions.md [hunk 2]
```diff
@@ -27,6 +27,16 @@ fun myFunction(arg1: Num, arg2: Num): Num {
 
 An interesting fact about functions is that they are not parsed unless they are used. This behavior exists because Amber allows you to omit specifying any type at all. When you use such function - then it generates different variants of this function with types that were used (without any duplications).
 
+You can use default parameters but requires to define the type of the parameter:
+
+```ab
+fun addition(a: Num, b: Num = 100): Num {
+    return a + b
+}
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:35]:
```suggestion
}

```

File: docs/basic_syntax/functions.md [hunk 3]
```diff
@@ -52,6 +62,22 @@ fun failing(name) {
 
 Notice that using `?` operator is automatically failing with the `status` code of the failing operation.
 
+You can mark a whole function as failable using `?` in the function declaration, an example:
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:65]:
```suggestion
If you specify the return type of a failable function, you must also append the `?` to the type name.
```

File: docs/basic_syntax/functions.md [hunk 4]
```diff
@@ -52,6 +62,22 @@ fun failing(name) {
...
+    if 0 > 5 {
+        fail 1
+    }
+
+    return 1
+}
+
+let a = failable() failed: echo "Failed"
+
+if a is Num: echo "Succeded"
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:78]:
```suggestion
```

File: docs/basic_syntax/functions.md [hunk 5]
```diff
@@ -52,6 +62,22 @@ fun failing(name) {
...
+```
+
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:80]:
```suggestion

Note that you cannot force a function to become failable by simply appending the `?` to the return type. The `?` can (and must) only be used in a function declaration, if the function is known to be failable.
```

File: docs/basic_syntax/functions.md (hunk 4 above)
Comment (Mte90) [reply docs/basic_syntax/functions.md:80]: why do you want to remove those lines?

Comment (mks-h) [reply docs/basic_syntax/functions.md:80]: They are unnecessary to the explanation and diverge attention.

Comment (Mte90) [reply docs/basic_syntax/functions.md:80]: In my opinion is a way to show how to use the function also if failable

Comment (mks-h) [reply docs/basic_syntax/functions.md:80]: That's totally unrelated to the discussed topic, though. The way failable functions are used is already demonstrated separately.

File: docs/basic_syntax/functions.md [hunk 6]
```diff
@@ -52,6 +64,21 @@ fun failing(name) {
...
+```ab
+fun failable(): Num? {
+    if 0 > 5 {
+        fail 1
+    }
+
+    return 1
+}
+
```
Comment (mks-h) [inline docs/basic_syntax/functions.md:77]:
```suggestion
```
A leftover newline, should be removed. Otherwise LGTM.

**Review by mks-h — APPROVED** [review]
Please apply the last suggested suggestion. After that, it can be merged.

**Review by mks-h — APPROVED** [review]

**Pull request merged**